src/main.rs

```diff
@@ -102,7 +89,7 @@ fn main() -> Result<(), aes_gcm_siv::Error> {
 
         // need to shorten it to u64 for now.
         // represent those 8 bytes as a single u64
-        let hash = <Fq as PrimeField>::from_le_bytes_mod_order(result[..8].try_into().unwrap());
+        let hash = <Fq as PrimeField>::from_le_bytes_mod_order(result[..].try_into().unwrap());
```

Comment:
what is scenario where this `try_into()` returns error

Reply:
when you do `result.try_into().unwrap()`